packages/browser/src/extensions/replay/external/lazy-loaded-session-recorder.ts

```diff
@@ -1188,7 +1194,7 @@ export class LazyLoadedSessionRecording implements LazyLoadedSessionRecordingInt
         }
 
         this._instance.persistence?.register({
-            [SESSION_RECORDING_IS_SAMPLED]: shouldSample,
+            [SESSION_RECORDING_IS_SAMPLED]: shouldSample ? sessionId : false,
```

Comment:
we were just storing boolean true in is sampled, and every so often we manage to carry on recording in a new session
let's just store the session id

Reply:
that way we can check if the _current_ session is sampled when deciding to record